### DISCUSSION TITLE
Scroll down pages is not fluid

### DISCUSSION THREAD

**Discussion by flyboy101** [infokiller/web-search-navigator#331] [General]
Please add a shortcut "f" for google finance (https://finance.google.com/finance)

Scrolling down search results is not fluid, as when a result at the bottom of the page is marked, the URL address and the google description basically lay on the page below. Page snapping should be improved.

Good job anyway!

**Comment by infokiller** [comment]
The google finance navigation was recently added.

Can you please check if the fluidity issue you see is still there, and if so add a screenshot showing the issue?

Closing for now, but please reopen if one of these issues is still present.

BTW, for future issue please open a separate issue for every problem you have (2 in this case IIUC).

**Comment by flyboy101** [comment]
Great job infokoller, one issue killed! I haven't found anything as smart as this extension so far, and I would be happy to give my little user-contribution.

Here you can download a series of screenshots I've made while just scrolling down the results with "arrow down" button: https://mega.nz/#F!mxFhDAJC!wUk3w0B6XLAYdTgoQ_eAFg

The behavior I referred to is visible in pic 5, 7, 10 and 14, where the focused result appears as 3rd, last, last and 3rd result in the showed list, respectively. I find this unpredictable behavior not very handy, and would rather prefer to have the focus always on the 1st position (or maybe 2nd) at every page snap in the research list. Here below the focus position pattern within the showed results in the page for every screenshot (scrolling down case):

pic 1  ---> 1
pic 2  ---> 2
pic 3  ---> 3
pic 4  ---> 4
----[ page snap ]--------
pic 5  ---> 1
pic 6  ---> 2
pic 7  ---> 3
----[ page snap ]--------
pic 8  ---> 1
pic 9  ---> 2
pic 10  ---> 3
pic 11  ---> 4
----[ page snap ]--------
pic 12  ---> 1
pic 13  ---> 2
pic 14  ---> 3

This would lead the attention of the user on the new set of results, without distractions from previously screened entries, furthermore with the maximum number of new fresh results in the screenpage. In addition, having the focus always on the top of the result list could turn out to be somehow more ergonomic, forcing the user to stretch up, rather than remaining "curved" on the bottom part of the result list.
I suspect something depend on the particular screen resolution one is using, though. Of course similar behavior should show up again while scrolling up the results, but with inverted pattern.

Here the behavior of some other similar extensions:
1. Google™ Search Shortcuts - RESULTER : apparently the focus get stuck to the first or the last result
2. Google Search Keyboard Shortcuts : seems to have similar behavior to your extension

I hope this may help. Feel free to drop me a line if you need a video for a clearer reproduction of the aforementioned effect.

**Comment by infokiller** [comment]
Thanks a lot @flyboy101 for the details! that helps a lot and now I understand the issue.
I agree the current behavior is not optimal and remaining on the first result in the visible list is better.
I hope I'll get time to tackle this soon, but realistically it may take O(months).

**Comment by flyboy101** [comment]
Welcome @infokiller. I dare the need of a result anchor in the viewport, but take it with a grain of salt.
Cheers.

**Comment by infokiller** [comment]
@flyboy101 this issue should be fixed on Chrome. You can already test it on master, but I will also do a release soon.

**Comment by flyboy101** [comment]
@infokiller  thanks for maintaining the project. I've testet v. 0.3.0 , but the behaviour is apparently the same: once you reach the last result in the viewport, the next "page down" sends the focus on some other results in the middle of the new viewport. The goal was to have the focus automatically pointing the first result of each viewport at the beginning.

**Comment by infokiller** [comment]
@flyboy101 this is only fixed in 0.3.2, so you should test that version.

**Comment by flyboy101** [comment]
I've tested v. 0.3.2. Now the cursor stays always on the bottom and the viewport moves accordingly. 
It would be better having the cursor on the top, then freely move it within the same viewport till the very last result is selected. In this last case the viewport should update to the next one with the cursor again on the top.

**Comment by infokiller** [comment]
I'm not sure I agree this is better (if I understand what you wrote correctly). Specifically, moving the viewport so that a result is at the top of the page will cause a big nonlinear scroll.
Why do you prefer the behavior you suggested? 
is it because you want to see the next results without having to navigate to them?
is it because the viewport moving is disturbing you?

Anyway, I think this can possibly be solved by adding two options:
1. `scroll_lookahead`- an integer specifying the number of results to "look ahead" when navigating. For example, if this option is set to 0 you will get the behavior in 0.3.2 where you will see the selected result at the bottom of the page (after navigating down enough). If this option is set to 1, the scrolling will begin when the selected result is the one before the bottom.
2. `scroll_size`- an integer specifying the number of results to scroll when needing to scroll because of a navigation to a result outside the viewport.

**Comment by flyboy101** [comment]
Yeap, you got the points:
1. Seeing all the possible results without navigating them would be advantageous when after a quick inspection by eye, you want to move directly to the next viewport (... maybe with another shortcut, e.g. by hitting "space bar" as already happen for normal web pages) as the first one contains too much ads, or just irrelevant results for you (still happen from time to time, even though you might want to change the keyword for a better search).
2. Continous scrollig of the viewport definitely strains the eyes in the long run, especially when your hardware is not that great.

Are there in your opinion any other faster way to do the job? I should here point out the fact that currently I've set to 50 the number of results per page in Google settings.

Actually v.0.3.2. does what I'm looking for, but on the opposite side, that is it focuses the cursor on the last entry rather than on the firts.  I would be more than happy to test a version with "scroll_lookahead" set to 1!

**Comment by infokiller** [comment]
> Are there in your opinion any other faster way to do the job?

What do you mean by the job? navigating a large number of results?

> I would be more than happy to test a version with "scroll_lookahead" set to 1!

From your description I actually think you want a large `scroll_size`. I will see if I have time to add these options and let you know

**Comment by elbotho** [comment]
@infokiller these options sound great.
Personally I don't mind the number of scrolls so much but always looking at the very bottom of the screen is quite unusual and gives me less change to scan ahead.
So 👍 and thank you so much, found this plugin today and it is going to change my life :)
